### PR TITLE
Support video file previews

### PR DESCRIPTION
**Pull request by MaxLeiter**
will rebase on #1806 if it's merged in first (assume it will be)

screen shot 2017-12-09 at 3 25 59 pm
screen shot 2017-12-09 at 3 25 52 pm

### CODE REVIEW

File: client/views/msg_preview.tpl [hunk 1]
```diff
@@ -5,15 +5,22 @@
 			<img src="{{thumb}}">
 		</a>
 	{{else}}
-		{{#if thumb}}
-			<a class="toggle-thumbnail" href="{{link}}" target="_blank" rel="noopener">
-				<img src="{{thumb}}" class="thumb">
+		{{#equal type "video"}}
```
Comment (YaManicKill) [inline client/views/msg_preview.tpl]: This file should probably start getting abstracted out if we are going to add a bunch of other ones to it. I know you are only adding 2 at this point, but it's going to get too big, I think.

File: src/plugins/irc-events/link.js [hunk 2]
```diff
@@ -106,6 +106,17 @@ function parse(msg, preview, res, client) {
 
 		break;
 
+	case "video/webm":
+	case "video/ogg":
+	case "video/mp4":
+		if (!preview.link.startsWith("https://")) {
+			break;
+		}
+		preview.res = res.type;
```
Comment (YaManicKill) [inline src/plugins/irc-events/link.js]: Hmmm, here's a question, why do we have 2 different types that we pass to the client?

Comment (xPaw) [reply src/plugins/irc-events/link.js]: One is "video" to know which template to render, one is full content-type like "video/mp4".

Comment (YaManicKill) [reply src/plugins/irc-events/link.js]: Ah right, so it's essentially 2 different levels of type of preview. 1 to decide which preview template to use, and another for the preview template (in this case, it's just passed to the video element, but the template could theoretically do something else with it). Makes sense.

**Comment by astorije** [comment]
👏

File: client/views/msg_preview.tpl (hunk 1 above)
Comment (MaxLeiter) [reply client/views/msg_preview.tpl]: Agreed. need to investigate handlebars `{{else if}}`, but I can't figure out how it works with helpers. For a future PR.

**Comment by YaManicKill** [comment]
Hmm, any reason you didn't merge, @astorije ? We have 2 approvals now.

Sadly, I can't approve for some reason as I get this, so someone else will have to.

File: client/views/msg_preview.tpl (hunk 1 above)
Comment (YaManicKill) [reply client/views/msg_preview.tpl]: Yeah, I wasn't meaning for this PR, it's just something we should do before we start adding third party site previews and stuff. The 2 PRs you have up are fine, it's not too big yet, it's just clearly **going** to get too big.

**Comment by astorije** [comment]
> any reason you didn't merge, @astorije ?

I just thought this PR required changes per @MaxLeiter's PR description, and also I didn't test it. Feel free to merge whenever you deem it ready. Great job @MaxLeiter!

**Comment by MaxLeiter** [comment]
Probably should be re-approved, changed layout of `msg_preview.tpl` quite a bit.

ping @astorije @YaManicKill

**Comment by YaManicKill** [comment]
Looks good, merging :-) thanks @MaxLeiter

**Pull request merged**